homeassistant/components/zamg/sensor.py

```diff
@@ -267,6 +265,7 @@ def __init__(
             configuration_url=MANUFACTURER_URL,
             name=coordinator.name,
         )
+        coordinator.api_fields = API_FIELDS
```

Comment:
What if another platform will be added in the future that requires another parameter than what the sensor platform needs? This design seems fragile. Why can't we fetch all available parameters with the coordinator by default?

Reply:
The thing is that the api is not the fastest to respond on a query.
With a reduction of the api parameters we get a faster response from the api server.

I know that that is not the ideal solution for that.